tests/data_out/data_out_12.cc

```diff
@@ -0,0 +1,173 @@
+                                             {1, 1, 0},
+                                             {0, 0, 1},
+                                             {1, 0, 1},
+                                             {0, 1, 1},
+                                             {1, 1, 1}};
+
+  const Quadrature<dim> support_quadrature(fe.get_unit_support_points());
+  FEValues<dim> fe_values(fe, support_quadrature, update_quadrature_points);
+  for (const auto cell : dof_handler.active_cell_iterators())
+    {
+      fe_values.reinit(cell);
+      const auto &qp = fe_values.get_quadrature_points();
+
+      for (unsigned int i = 0; i < dofs_per_cell; ++i)
+        {
+          const auto &       this_qp = qp[i];
+          const unsigned int i_group = fe.system_to_base_index(i).first.first;
+          const unsigned int i_comp  = fe.system_to_component_index(i).first;
+          if (i_group == 0)
+            // vector
+            {
+              if (i_comp == 0)
+                v(i) = 1;
+              else if (i_comp == 1)
+                v(i) = this_qp[1];
+              else
+                v(i) = 0.;
+            }
+          else
+            // tensor
+            {
+              // find matching location
+              unsigned int p = 0;
+              for (; p < 9; ++p)
+                {
+                  Point<dim> point(coords[p][0], coords[p][1], coords[p][2]);
+                  const auto diff = point - this_qp;
+                  if (diff.norm() < 1e-10)
+                    break;
+                }
+
+              const unsigned int tens_comp = i_comp - dim;
+              const unsigned int ii =
+                Tensor<2, dim>::unrolled_to_component_indices(tens_comp)[0];
+              const unsigned int jj =
+                Tensor<2, dim>::unrolled_to_component_indices(tens_comp)[1];
+              v(i) = tensors[p][ii][jj];
+            }
+        }
+    }
+
+  std::vector<DataComponentInterpretation::DataComponentInterpretation>
+    data_component_interpretation(
+      dim + dim * dim,
+      DataComponentInterpretation::component_is_part_of_tensor);
+  std::fill(data_component_interpretation.begin(),
+            data_component_interpretation.begin() + dim,
+            DataComponentInterpretation::component_is_part_of_vector);
+
+  std::vector<std::string> component_name(dim + dim * dim, "tensor");
+  std::fill(component_name.begin(), component_name.begin() + dim, "vector");
+
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(v,
+                           component_name,
+                           DataOut<dim>::type_dof_data,
+                           data_component_interpretation);
+  data_out.build_patches();
+
+  /*
+  std::ofstream out("output_" + Utilities::int_to_string(dim) + "d.vtu");
+  data_out.write_vtu(out);
+  */
+
+  data_out.write_vtu(deallog.get_file_stream());
+}
+
+
+
+int
+main()
+{
+  initlog();
+  check<3>();
```

Comment:
Have you also checked that the 2d output is readable by Paraview? I think that its probably a good idea to verify this.